### ISSUE TITLE
Standard Price Is Not Selected Security Query False Positive

### ISSUE THREAD

**Issue by cxMiguelSilva** [Checkmarx/kics#5876]
### Actual Behavior

`Standard Price Is Not Selected` query resulted in a false positive in line 219, where `[parameters('virtualMachineTier')]` is not being recognized as  the standard price. 
[Documentation](https://learn.microsoft.com/en-us/azure/templates/microsoft.security/pricings?tabs=json&pivots=deployment-language-arm-template#pricingproperties-1)

KICS Scan Result:
```
{
	"query_name": "Standard Price Is Not Selected",
	"query_id": "2081c7d6-2851-4cce-bda5-cb49d462da42",
	"query_url": "https://docs.microsoft.com/en-us/azure/templates/microsoft.security/pricings?tabs=json#pricingproperties-object",
	"severity": "MEDIUM",
	"platform": "AzureResourceManager",
	"category": "Networking and Firewall",
	"description": "Azure Security Center provides more features for standard pricing mode, so it must be activated.",
	"description_id": "2081c7dc",
	"cis_description_id": "CIS Security - CIS Microsoft Azure Foundations Benchmark v1.3.1 - Rule 2.2",
	"cis_description_title": "Ensure that Azure Defender is set to On for App Service",
	"cis_description_text": "Turning on Azure Defender enables threat detection for App Service, providing threat intelligence, anomaly detection, and behavior analytics in the Azure Security Center.\nEnabling Azure Defender for App Service allows for greater defense-in-depth, with threat detection provided by the Microsoft Security Response Center (MSRC).",
	"files": [
		{
			"file_name": "../../path/azuredeploy.json",
			"similarity_id": "78ce032f8a7dbf64d319a72db8ea37fb6efe4a1593db0a8f0beed2654977f1f9",
			"line": 219,
			"resource_type": "Microsoft.Security/pricings",
			"resource_name": "AppServices",
			"issue_type": "IncorrectValue",
			"search_key": "resources.name=AppServices.properties.pricingTier",
			"search_line": 0,
			"search_value": "",
			"expected_value": "'pricingTier' should be set to standard",
			"actual_value": "'pricingTier' is set to [parameters('appServiceTier')]"
		},
		...
```

### Steps to Reproduce the Problem

(Command line arguments and flags used)

1. Run KICS Scan command
`docker run -v /path/to/sample.yml:/path checkmarx/kics:v1.6.1 scan -p /path`

Sample to reproduce the problem:
```
{
    "$schema": "https://schema.management.azure.com/schemas/2019-04-01/deploymentTemplate.json#",
    "contentVersion": "1.0.0.1",
    "parameters": {
        "workspaceName": {
            "type": "string",
            "defaultValue": "az-security-workspace",
            "allowedValues": [
                "az-security-workspace"
            ],
            "metadata": {
                "description": "Name of the central Log Analytics workspace that stores security event and data collected by Azure Security Center"
            }
        },
        "workspaceRgName": {
            "type": "string",
            "defaultValue": "azsec-security-rg",
            "allowedValues": [
                "azsec-security-rg"
            ],
            "metadata": {
                "description": "Name of the resource group where the central log analytics workspace belongs to"
            }
        },
        "autoProvisionSetting": {
            "type": "string",
            "defaultValue": "On",
            "allowedValues": [
                "On",
                "Off"
            ],
            "metadata": {
                "description": "Specify whether Auto Provisoning is turned on or off"
            }
        },
        "ascOwnerEmail": {
            "type": "string",
            "metadata": {
                "description": "Email of the administrator who should be notified about Azure Security Center alert"
            }
        },
        "ascOwnerContact": {
            "type": "string",
            "metadata": {
                "description": "Phone number of the administrator should be notified about Azure Security Center alert"
            }
        },
        "highSeverityAlertNotification": {
            "type": "string",
            "defaultValue": "On",
            "allowedValues": [
                "On",
                "Off"
            ],
            "metadata": {
                "description": "Specify whether you want to notify high severity alert to ASC administrator"
            }
        },
        "subscriptionOwnerNotification": {
            "type": "string",
            "defaultValue": "On",
            "allowedValues": [
                "On",
                "Off"
            ],
            "metadata": {
                "description": "Specifiy whether you want to notify high severity alert to subscription owner"
            }
        },
        "virtualMachineTier": {
            "type": "string",
            "defaultValue": "Standard",
            "allowedValues": [
                "Standard",
                "Free"
            ],
            "metadata": {
                "description": "Specifiy whether you want to enable Standard tier for Virtual Machine resource type"
            }
        },
        "appServiceTier": {
            "type": "string",
            "defaultValue": "Standard",
            "allowedValues": [
                "Standard",
                "Free"
            ],
            "metadata": {
                "description": "Specify whether you want to enable Standard tier for Azure App Service resource type"
            }
        },
        "paasSQLServiceTier": {
            "type": "string",
            "defaultValue": "Standard",
            "allowedValues": [
                "Standard",
                "Free"
            ],
            "metadata": {
                "description": "Specify whether you want to enable Standard tier for PaaS SQL Service resource type"
            }
        },
        "sqlServerOnVmTier": {
            "type": "string",
            "defaultValue": "Standard",
            "allowedValues": [
                "Standard",
                "Free"
            ],
            "metadata": {
                "description": "Specify whether you want to enable Standard tier for SQL Server on VM resource type"
            }
        },
        "storageAccountTier": {
            "type": "string",
            "defaultValue": "Standard",
            "allowedValues": [
                "Standard",
                "Free"
            ],
            "metadata": {
                "description": "Specify whether you want to enable Standard tier for Storage Account resource type"
            }
        },
        "kubernetesServiceTier": {
            "type": "string",
            "defaultValue": "Standard",
            "allowedValues": [
                "Standard",
                "Free"
            ],
            "metadata": {
                "description": "Specify whether you want to enable Standard tier for Kubernetes service resource type"
            }
        },
        "containerRegistryTier": {
            "type": "string",
            "defaultValue": "Standard",
            "allowedValues": [
                "Standard",
                "Free"
            ],
            "metadata": {
                "description": "Specify whether you want to enable Standard tier for Container Registry resource type"
            }
        },
        "keyvaultTier": {
            "type": "string",
            "defaultValue": "Standard",
            "allowedValues": [
                "Standard",
                "Free"
            ],
            "metadata": {
                "description": "Specify whether you want to enable Standard tier for Key Vault resource type"
            }
        },
        "integrationName": {
            "type": "string",
            "allowedValues": [
                "MCAS",
                "MDATP"
            ],
            "metadata": {
                "description": "Select integration name to enable. Only MCAS or MDATP is supported."
            }
        },
        "integrationEnabled": {
            "type": "bool",
            "allowedValues": [
                true,
                false
            ],
            "metadata": {
                "description": "Specify whether you want to enable or not."
            }
        }
    },
    "resources": [
        {
            "type": "Microsoft.Security/workspaceSettings",
            "apiVersion": "2017-08-01-preview",
            "name": "default",
            "properties": {
                "scope": "[subscription().id]",
                "workspaceId": "[concat(subscription().id,'/resourceGroups/',parameters('workspaceRgName'),'/providers/Microsoft.OperationalInsights/workspaces/',parameters('workspaceName'))]"
            }
        },
        {
            "type": "Microsoft.Security/autoProvisioningSettings",
            "apiVersion": "2017-08-01-preview",
            "name": "default",
            "properties": {
                "autoProvision": "[parameters('autoProvisionSetting')]"
            }
        },
        {
            "type": "Microsoft.Security/securityContacts",
            "apiVersion": "2017-08-01-preview",
            "name": "default1",
            "properties": {
                "emails": "[parameters('ascOwnerEmail')]",
                "phone": "[parameters('ascOwnerContact')]",
                "alertNotifications": {
                    "state": "On",
                    "minimalSeverity": "[parameters('highSeverityAlertNotification')]"
                },
                "notificationsByRole": {
                    "state": "On",
                    "roles": "[parameters('subscriptionOwnerNotification')]"
                }
            }
        },
        {
            "type": "Microsoft.Security/pricings",
            "apiVersion": "2018-06-01",
            "name": "VirtualMachines",
            "properties": {
                "pricingTier": "[parameters('virtualMachineTier')]"
            }
        },
        {
            "type": "Microsoft.Security/pricings",
            "apiVersion": "2018-06-01",
            "name": "AppServices",
            "dependsOn": [
                "[concat('Microsoft.Security/pricings/VirtualMachines')]"
            ],
            "properties": {
                "pricingTier": "[parameters('appServiceTier')]"
            }
        },
        {
            "type": "Microsoft.Security/pricings",
            "apiVersion": "2018-06-01",
            "name": "SqlServers",
            "dependsOn": [
                "[concat('Microsoft.Security/pricings/AppServices')]"
            ],
            "properties": {
                "pricingTier": "[parameters('paasSQLServiceTier')]"
            }
        },
        {
            "type": "Microsoft.Security/pricings",
            "apiVersion": "2018-06-01",
            "name": "SqlServerVirtualMachines",
            "dependsOn": [
                "[concat('Microsoft.Security/pricings/SqlServers')]"
            ],
            "properties": {
                "pricingTier": "[parameters('sqlServerOnVmTier')]"
            }
        },
        {
            "type": "Microsoft.Security/pricings",
            "apiVersion": "2018-06-01",
            "name": "StorageAccounts",
            "dependsOn": [
                "[concat('Microsoft.Security/pricings/SqlServerVirtualMachines')]"
            ],
            "properties": {
                "pricingTier": "[parameters('storageAccountTier')]"
            }
        },
        {
            "type": "Microsoft.Security/pricings",
            "apiVersion": "2018-06-01",
            "name": "KubernetesService",
            "dependsOn": [
                "[concat('Microsoft.Security/pricings/StorageAccounts')]"
            ],
            "properties": {
                "pricingTier": "[parameters('kubernetesServiceTier')]"
            }
        },
        {
            "type": "Microsoft.Security/pricings",
            "apiVersion": "2018-06-01",
            "name": "ContainerRegistry",
            "dependsOn": [
                "[concat('Microsoft.Security/pricings/KubernetesService')]"
            ],
            "properties": {
                "pricingTier": "[parameters('containerRegistryTier')]"
            }
        },
        {
            "type": "Microsoft.Security/pricings",
            "apiVersion": "2018-06-01",
            "name": "KeyVaults",
            "dependsOn": [
                "[concat('Microsoft.Security/pricings/ContainerRegistry')]"
            ],
            "properties": {
                "pricingTier": "[parameters('keyvaultTier')]"
            }
        },
        {
            "type": "Microsoft.Security/settings",
            "apiVersion": "2019-01-01",
            "name": "[parameters('integrationName')]",
            "kind": "DataExportSettings",
            "properties": {
                "enabled": "[parameters('integrationEnabled')]"
            }
        }
    ],
    "outputs": {
    }
}
```
